src/mpid/ch4/generic/am/mpidig_am_send.h

```diff
@@ -54,13 +55,13 @@ MPL_STATIC_INLINE_PREFIX int mpidig_eager_limit(int is_local)
 MPL_STATIC_INLINE_PREFIX int MPIDIG_do_eager_send(const void *buf, MPI_Aint count,
                                                   MPI_Datatype datatype, int rank, int tag,
                                                   MPIR_Comm * comm, int context_offset,
-                                                  MPIDI_av_entry_t * addr, MPIR_Request ** request,
-                                                  MPIR_Errflag_t errflag);
+                                                  MPIDI_av_entry_t * addr, uint8_t flags,
+                                                  MPIR_Request ** request, MPIR_Errflag_t errflag);
```

Comment:
If we create the request object in `MPIDIG_isend_impl`, then we can directly embed the flags in the request, thus simplifying the interface and also reduce the code duplication a bit. What do you think?

Reply:
> If we create the request object in `MPIDIG_isend_impl`, then we can directly embed the flags in the request, thus simplifying the interface and also reduce the code duplication a bit. What do you think?
> 
> PS: does the request object have the fields for `msg_hdr` ? If it does, then maybe we can simply init the `am_hdr` in the request, then directly send it, to save some duplication later. I think have some fields in the send-side request can facilitate later protocol handling if needed, e.g. pipeline or rdma.

Almost. The request does not have the error bits or flags fields now, but could easily be added. This is probably a good idea, especially if we can avoid requiring the lower layer having to copy the header.

Reply:
Yes. We do need the flags in the request later. I will take care of that in the later PIPELINE PR. I think we can keep the current form in this PR. As for putting the message into the request, I think it is generally OK.